designs/_posts/2017-07-13-improved-command-line-reporting.md

```diff
@@ -0,0 +1,530 @@
+---
+layout: contribute
+title: Improved Command Line Reporting
+---
+
+# Design Document: Improved command line reporting in Bazel
+
+**Design documents are not descriptions of the current functionality of Bazel.
+Always go to the documentation for current information.**
+
+**Status**: In Progress.
+
+**Author**: [Chloe Calvarin](mailto:ccalvarin@google.com)
+
+**Design document published**: 13 July 2017
+
+## Background
+
+Currently, there is no good way for a Bazel user to get the command line that 
+initiated a build, if they do not have access to the terminal history. The 
+GotOptionsEvent tracks the various types of options (notably, startup and 
+command options) and this information is starting to be used in Bazel by the 
+BuildEventProtocol (BEP, see the proto definition in 
+//src/main/java/com/google/devtools/build/lib/buildeventstream/proto/build_event_stream.proto, 
+for context). However, this information is incomplete, and, unfortunately, 
+not entirely correct.
+
+The output information in the BEP is currently as follows:
+```
+message OptionsParsed {
+  repeated string startup_options = 1;
+  repeated string explicit_startup_options = 2;
+  repeated string cmd_line = 3;
+  repeated string explicit_cmd_line = 4;
+  blaze.invocation_policy.InvocationPolicy invocation_policy = 5;
+}
+```
+
+Startup options, the options that go before the command, are listed in two ways:
+
++ `startup_options` : The effective options, both those mentioned explicitely 
+   and implicitely (say, in a bazelrc)
++ `explicit_startup_options` : The explicit options, only those mentioned 
+   explicitely on the command line.
+
+As are the command options:
+
++ `cmd_line` : The effective options. These include those listed explicitely 
+   in the command line, but also have expansion flags expanded, those mentioned
+   in bazelrc's, and any alterations from the invocation policy.
++ `explicit_cmd_line` : The explicit options listed on the command line.
+
+This is insufficient information to mirror the full user input. The executable,
+command, and targets are listed elsewhere, and the parts, if recombined, do not
+accurately reflect what was provided by the user, since they were never 
+intended to be matching parts of the whole. It is also not fully correct. 
+
+## Goals
+
+Basically, we want the following: the "literal" command line that was passed 
+by the user, and a "reproducible" command line, along with the amorphous 
+request of "any additional information that might help debug an issue". 
+
+Unfortunately, reproducibility is a red herring, as it can mean many different
```

Comment:
"red herring" may not be understood by everyone  -- suggest using "distraction" or something like "reproducibility cannot be our focus"